packages/server/graphql/intranetSchema/mutations/runScheduledJobs.ts

```diff
@@ -35,28 +30,52 @@ const processMeetingStageTimeLimits = async (job: ScheduledJobMeetingStageTimeLi
       .nth(0)
       .default(null) as unknown) as SlackAuth
   }).run()
+  return {slackNotification, slackAuth}
+}
 
-  let sendViaSlack = Boolean(slackAuth?.botAccessToken && slackNotification?.channelId)
-  if (sendViaSlack) {
-    const {channelId} = slackNotification
-    if (!channelId) {
-      sendViaSlack = false
-    } else {
-      const {botAccessToken} = slackAuth
-      const manager = new SlackServerManager(botAccessToken)
-      const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`)
-      const slackText = `Time’s up! Advance your meeting to the next phase: ${meetingUrl}`
-      const res = await manager.postMessage(channelId, slackText)
-      if (!res.ok) {
-        sendViaSlack = false
-      }
-    }
+const processMeetingStageTimeLimits = async (
+  job: ScheduledJobMeetingStageTimeLimit,
+  {dataLoader}
+) => {
+  // get the meeting
+  // get the facilitator
+  // see if the facilitator has turned on slack notifications for the meeting
+  // detect integrated services
+  // if slack, send slack
+  // if mattermost, send mattermost
+  // if no integrated notification services, send an in-app notification
+  const {meetingId} = job
+  const meeting = (await dataLoader.get('newMeetings').load(meetingId)) as Meeting
+  const {teamId, facilitatorUserId} = meeting
+  const {slackNotification, slackAuth} = await getSlackNotificationAndAuth(
+    teamId,
+    facilitatorUserId
+  )
+  const mattermostAuth = await dataLoader.get('mattermostAuthByTeamId').load(teamId)
+  const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`)
+
+  let sendViaSlack = false
+  let sendViaMattermost = mattermostAuth?.isActive
+
+  if (slackAuth?.botAccessToken && slackNotification?.channelId) {
+    sendViaSlack = true
+    const manager = new SlackServerManager(slackAuth.botAccessToken)
+    const slackText = `Time’s up! Advance your meeting to the next phase: ${meetingUrl}`
+    const res = await manager.postMessage(slackNotification.channelId, slackText)
```

Comment:
this is a hot take & feel free to fight me

I think both SlackManager & MattermostManager should implement an abstract class called NotificationManager.

NotificationManager has abstract methods like `timeLimitEnd(meetingId, teamId, dataLoader)`
then, we can do things like this:

```ts
sendViaMattermost && mattermostManager.timeLimitEnd()
sendViaSlack && slackManager.timeLimitEnd()
```

Reply:
if we wanted to get really fancy, then we could have `abstract class NotificationManager` have statics like 

```
static DefaultTimeLimitEndMessage = (meetingUrl) => `Time up for ${meetingUrl}`
```

Reply:
Finally, we might have a `ServiceNotifier` class that does the following:
- loops through all notification integrations & creates NotificationManagers for each valid one.
- This also implements the `NotificationManager` interface.
- End result, you call `notificationManager.timeLimitEnd()` and that calls slack & mattermost's implementations of that same method.

Reply:
YAAAAS! I want to do this so badly. I decided to stop short in this PR because I have a similar proposal for this pattern that includes rolling up default oauth2 methods into the abstract class. I'd like to refactor the existing Slack & Mattermost integrations to take advantage.

Since I've already started this work in another branch, I don't want to pull all that into this PR.